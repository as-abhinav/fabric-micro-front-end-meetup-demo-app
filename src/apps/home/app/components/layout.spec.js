import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import HeroLayout from './layout';

describe('Layout', () => {
  const node = mount(
    <HeroLayout backgroundClassName="something">
      <div className="child-component">blah child</div>
    </HeroLayout>
  );
  it('should render layout component', () => {
    expect(node.find('section').length).toEqual(1);
    expect(node.find('img').length).toEqual(1);
    expect(node.find('.child-component').length).toEqual(1);
  });

  it('should match snapshot', () => {
    const component = renderer.create(
      <HeroLayout backgroundClassName="something">
        <div>blah child</div>
      </HeroLayout>
    );
    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
