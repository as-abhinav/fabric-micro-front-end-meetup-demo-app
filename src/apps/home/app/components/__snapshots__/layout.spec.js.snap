// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should match snapshot 1`] = `
<section
  className="HeroLayout-root-1"
>
  <div
    className="MuiContainer-root HeroLayout-container-2 MuiContainer-maxWidthLg"
  >
    <img
      alt="wonder"
      height="80"
      src="https://uploads-ssl.webflow.com/5d8ac4a88c12c797c7368ab7/5d9d4e6edf1e449b647b1ac0_FabricLogo.svg"
      width="147"
    />
    <div>
      blah child
    </div>
    <div
      className="HeroLayout-backdrop-3"
    />
    <div
      className="HeroLayout-background-4 something"
    />
  </div>
</section>
`;
